app/V1Module/presenters/AsyncJobsPresenter.php

```diff
@@ -0,0 +1,156 @@
+        $this->sendSuccessResponse($asyncJob);
+    }
+
+    public function checkList()
+    {
+        if (!$this->asyncJobsAcl->canList()) {
+            throw new ForbiddenRequestException("You cannot list async jobs");
+        }
+    }
+
+    /**
+     * Retrieves details about async jobs that are either pending or were recently completed.
+     * @GET
+     * @param int|null $ageThreshold Maximal time since completion (in seconds), null = only pending operations
+     * @param bool|null $includeScheduled If true, pending scheduled events will be listed as well
+     * @throws BadRequestException
+     */
+    public function actionList(?int $ageThreshold, ?bool $includeScheduled)
+    {
+        if ($ageThreshold && $ageThreshold < 0) {
+            throw new BadRequestException("Age threshold must not be negative.");
+        }
+
+        // criterium for termination (either pending or within threshold)
+        $terminatedAt = Criteria::expr()->eq('terminatedAt', null);
+        if ($ageThreshold) {
+            $thresholdDate = new DateTime();
+            $thresholdDate->modify("-$ageThreshold seconds");
+            $terminatedAt = Criteria::expr()->orX(
+                $terminatedAt,
+                Criteria::expr()->gte('terminatedAt', $thresholdDate)
+            );
+        }
+
+        $criteria = Criteria::create()->where(
+            $includeScheduled
+                ? $terminatedAt
+                : Criteria::expr()->andX(
+                    $terminatedAt,
+                    Criteria::expr()->eq('scheduledAt', null)
+                )
+        );
+        $criteria->orderBy([ 'createdAt' => 'ASC' ]);
+        $jobs = $this->asyncJobs->matching($criteria)->toArray();
+
+        $jobs = array_filter($jobs, function ($job) {
+            return $this->asyncJobsAcl->canViewDetail($job);
+        });
+
+        $this->sendSuccessResponse($jobs);
+    }
+
+    public function checkAbort(string $id)
+    {
+        $asyncJob = $this->asyncJobs->findOrThrow($id);
+        if (!$this->asyncJobsAcl->canAbort($asyncJob)) {
+            throw new ForbiddenRequestException("You cannot abort selected async job");
+        }
+    }
+
+    /**
+     * Retrieves details about particular async job.
+     * @POST
+     * @param string $id job identifier
+     * @throws NotFoundException
+     */
+    public function actionAbort(string $id)
+    {
+        $this->asyncJobs->beginTransaction();
+        try {
+            $asyncJob = $this->asyncJobs->findOrThrow($id);
+            if ($asyncJob->getStartedAt() === null && $asyncJob->getTerminatedAt() === null) {
+                // if the job has not been started yet, it can be aborted
+                $asyncJob->setTerminatedNow();
+                $asyncJob->appendError("ABORTED");
+                $this->asyncJobs->persist($asyncJob);
+                $this->asyncJobs->commit();
+            } else {
+                $this->asyncJobs->rollback();
```

Comment:
Why rollback only read?

Reply:
Since it might be faster for the database (this intel is not verified though).